apis/apps/v1alpha1/type.go

```diff
@@ -115,9 +122,77 @@ type LegacyRenderedTemplateSpec struct {
 	ConfigTemplateExtension `json:",inline"`
 }
 
+type ComponentParameters struct {
+	// Specifies the config file name.
+	//
+	// It must be a string of maximum 63 characters, and can only include lowercase alphanumeric characters,
+	// hyphens, and periods.
+	// The name must start and end with an alphanumeric character.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern:=`^[a-z0-9]([a-z0-9\.\-]*[a-z0-9])?$`
+	Name string `json:"name"`
```

Comment:
There's no need to bind parameters to config files. Instead, we should define a list of parameters (key-value pairs) for a Component in the ComponentDefinition. This way, users can set these parameters for the Component directly without specifying the config files in which they reside.

This approach is preferable for two reasons:

It reduces the burden on users by eliminating the need to specify config files when setting parameters.
Some parameters don't appear in any config files at all.